docs/miscellaneous.rst

```diff
@@ -303,6 +303,124 @@ The following is the order of precedence for operators, listed in order of evalu
+
+.. warning::
+    Solidity does not automatically detect over/underflow.
+
+Additionally, the EVM has its own overflow semantics for results that exceed 256 bits. See: `The Ethereum Yellow Paper <https://ethereum.github.io/yellowpaper>`_, Appendix H, Section 2 (Instruction set).
+
+Examples:
+
+``uint8(257) == 257 mod 2**8 == 1``
+
+``uint8(2 - 5) == -3 mod 2**8 == 256 - 3 == 253``
+
+``int8(-129) == -129 mod 256 == 127``
+
+**Casting and demotion**
+
+When casting a value into an integer type, the result is always reduced `modulo N`, where `N` is the bit-size of that integer type. This is important to remember when demoting an integer to a new type with a smaller bit-size.
+
+Example: ``uint8(257) = 1``;
+
+**Special cases**
+
+The range of signed integer types, i.e. ``[-2**(N - 1), 2**(N - 1) - 1]``, is asymmetric around 0. As a result of this, inverting the minimum value will result in a value that is ``1`` greater then the maximum value, making it wrap around back to itself:
+
+``-INT_MIN = -(-2**(N - 1)) = 2**(N - 1) = 2**(N - 1) - 1 + 1 = INT_MAX + 1 = INT_MIN``
+
+Table I: Rules for operators and unsigned Integers
+--------------------------------------------------
+
++------------+----------+-----------+---------+
+|  Operator  | Overflow | Underflow | Notes   |
++============+==========+===========+=========+
+| +, ++, +=  | x        |           |         |
++------------+----------+-----------+---------+
+| \+ (unary) | N/A      | N/A       | 1       |
++------------+----------+-----------+---------+
+| -, --, -=  |          | x         |         |
++------------+----------+-----------+---------+
+| \- (unary) |          |           | 2       |
++------------+----------+-----------+---------+
+| \*, \*=    | x        |           |         |
++------------+----------+-----------+---------+
+| \*\*       | x        |           | 3       |
++------------+----------+-----------+---------+
+| /, /=      |          |           |         |
++------------+----------+-----------+---------+
+| %, %=      |          |           |         |
++------------+----------+-----------+---------+
+| <<, <<=    |          |           | 4       |
++------------+----------+-----------+---------+
+| >>, >>=    |          |           | 5       |
++------------+----------+-----------+---------+
+| &, &=      |          |           |         |
++------------+----------+-----------+---------+
+| \|, \|=    |          |           |         |
++------------+----------+-----------+---------+
+| ^, ^=      |          |           |         |
++------------+----------+-----------+---------+
+| ~, ~=      |          |           |         |
++------------+----------+-----------+---------+
+
+1) Unary ``+`` has been deprecated.
+2) Unary ``-`` works on ``uints``, e.g. ``-uint(1) == ~uint(0)``.
```

Comment:
_also_ works on uints.

Reply:
Perhaps we should also tell users of invariants: `-x` is always identical to `0 - x`. `a += b` is always identical to `a = a + b` (with the difference that a is only evaluated once) and so on.

Reply:
first suggestion is fixed in the next push. 

i will push without any of those invariants, so that is the only remaining issue (since it seems like a pretty big one).